contracts/InverseVesterFactory.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.3;
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IInv, InverseVester } from "./InverseVester.sol";
+
+/**
+ * @title Inverse token vesting factory contract
+ * @author Inverse Finance
+ * @notice Factory contract to create configurable vesting agreement
+ */
+contract InverseVesterFactory is Ownable {
+    /// @dev Emitted when vesting starts
+    event VestingCreated(address recipient, address inverseVester, uint256 amount, uint16 vestingDuration);
+
+    /// @dev Inverse finance governance timelock
+    address public timelock;
+    /// @dev Inverse finance treasury token
+    address public immutable inv;
+    /// @dev Registry of vesting agreements by recipient
+    mapping(address => InverseVester[]) public inverseVestersByRecipient;
```

Comment:
Might be interesting to keep track of all recipients with an `address[]`. That way it's easy to grab the entire list of recipients.

Could then remove the `allInverseVesters` member. Getting all vesting contracts could be done by looping through all recipients and then all looping through all their vesting contracts.

Reply:
Yeah I thought about that. I'll need a set then